apps/api/src/app/auth/usecases/register/user-register.command.ts

```diff
@@ -9,18 +11,34 @@ export class UserRegisterCommand extends BaseCommand {
   email: string;
 
   @IsDefined()
+  @IsString()
   @MinLength(8)
   password: string;
 
   @IsDefined()
+  @IsString()
   firstName: string;
 
   @IsOptional()
+  @IsString()
   lastName?: string;
 
   @IsOptional()
+  @IsString()
   organizationName?: string;
 
   @IsOptional()
+  @IsEnum(SignUpOriginEnum)
   origin?: SignUpOriginEnum;
+
+  @IsOptional()
+  @IsEnum(JobTitleEnum)
+  jobTitle?: JobTitleEnum;
+
+  @IsString()
+  @IsOptional()
```

Comment:
I think `IsOptional` should come before `IsString` here

Reply:
Okay I see precedence doesn't matter

Reply:
Good point, i think in both cases, IsOptional will tell class-validator to skip validation, however, it will be more optimized to check IsOptional first and then skip the next decorator.

Made a small Google search and saw this comment from one of the members
https://github.com/typestack/class-validator/issues/1688#issuecomment-1353943121, So i wonder if they are supporting order at all.